lib/common/ipc.c

```diff
@@ -278,6 +278,58 @@ crm_client_disconnect_all(qb_ipcs_service_t *service)
     }
 }
 
+/*!
+ * \internal
+ * \brief Allocate a new crm_client_t object based on an IPC connection
+ *
+ * \param[in] c           IPC connection (or NULL to allocate generic client)
+ * \param[in] key         Connection table key (or NULL to use sane default)
+ * \param[in] uid_client  UID corresponding to c (ignored if c is NULL)
+ *
+ * \return Pointer to new crm_client_t (or NULL on error)
+ */
+static crm_client_t *
+client_from_connection(qb_ipcs_connection_t *c, void *key, uid_t uid_client)
+{
+    crm_client_t *client = calloc(1, sizeof(crm_client_t));
+
+    if (client == NULL) {
+        crm_perror(LOG_ERR, "Allocating client");
+        return NULL;
+    }
+
+    if (c) {
+#if ENABLE_ACL
+        client->user = uid2username(uid_client);
+        if (client->user == NULL) {
+            client->user = strdup("#unprivileged");
+            CRM_CHECK(client->user != NULL, free(client); return NULL);
+            crm_err("Unable to enforce ACLs for user ID %d, assuming unprivileged",
+                    uid_client);
```

Comment:
`uid_t` is `unsigned int` on my machine

Reply:
Good catch, I missed that when updating get_shadow_file() / uid2username() and basing on those. POSIX only limits uid_t to an integer type. I think %llu with a cast would be appropriate.